examples/OSMPDummySensor/OSMPDummySensor.cpp

```diff
@@ -236,15 +236,28 @@ fmi2Status COSMPDummySensor::doExitInitializationMode()
     return fmi2OK;
 }
 
-void rotatePoint(double x, double y, double z,double yaw,double pitch,double roll,double &rx,double &ry,double &rz)
+
+/*!
+     * \brief 3D rotation matrix function. Calculates local X,Y,Z coordinates relative from reference.  
+     * \param x Global relative x-coordinate.
+     * \param y Global relative y-coordinate.
+     * \param z Global relative z-coordinate
+     * \param yaw Global yaw reference of rotation.
+     * \param pitch Global pitch reference of rotation.
+     * \param roll Global roll reference of rotation.
+     * \param rx Local x-coordinate after rotation. 
+     * \param ry Local y-coordinate after rotation.
+     * \param rz Local z-coordinate after rotation.
+     */ 
+void rotatePoint3D(double x, double y, double z,double yaw,double pitch,double roll,double &rx,double &ry,double &rz) // TODO: Find out why 3D rotation doesn't work. Using 2D rotation instead.
 {
     double matrix[3][3];
-    double cos_yaw = cos(yaw);
-    double cos_pitch = cos(pitch);
-    double cos_roll = cos(roll);
-    double sin_yaw = sin(yaw);
-    double sin_pitch = sin(pitch);
-    double sin_roll = sin(roll);
+    double cos_yaw = cos(yaw*3.14/180);

```

Comment:
Recommend using `M_PI` for a better approximation. It should be defined as standard.

Reply:
Fixed